common_app/main.c

```diff
@@ -0,0 +1,85 @@
+/*****************************************************************************
+ *   (c) 2020 Ledger SAS.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *****************************************************************************/
+
+#include <stdint.h>  // uint*_t
+#include <string.h>  // memset, explicit_bzero
+
+#include "os.h"
+#include "io.h"
+
+ux_state_t G_ux;
+bolos_ux_params_t G_ux_params;
+
+/**
+ * Exit the application and go back to the dashboard.
+ */
+WEAK void app_exit() {
+    BEGIN_TRY_L(exit) {
+        TRY_L(exit) {
+            os_sched_exit(-1);
+        }
+        FINALLY_L(exit) {
+        }
+    }
+    END_TRY_L(exit);
+}
+
+/**
+ * Main loop to setup USB, Bluetooth, UI and launch app_main().
+ */
+WEAK __attribute__((section(".boot"))) int main() {
+    __asm volatile("cpsie i");
+
+    os_boot();
+
+    for (;;) {
+        // Reset UI
+        memset(&G_ux, 0, sizeof(G_ux));
```

Comment:
should be replaced with UX_INIT();
See https://github.com/LedgerHQ/app-boilerplate/pull/55